### DISCUSSION TITLE
How to Put a Custom Widget into a Custom Menu

### DISCUSSION THREAD

**Discussion by sboudreaux96** [eclipse-theia/theia#9622] [Q&A] [answered]
Hello - I have a custom menu item I created next to File, Edit, etc. (it is just like sample-menu from the api-samples) and I wanted to know how I would go about adding my custom widget to that custom menu?

**Comment by vince-fugnitto** [comment, marked as answer]
@sboudreaux96 use the API to register menus (https://eclipse-theia.github.io/theia/docs/next/classes/core.menumodelregistry-1.html) like in the example:

https://github.com/eclipse-theia/theia/blob/6afae9e443a09713a69733d8e85c32029639ae11/examples/api-samples/src/browser/menu/sample-menu-contribution.ts#L48-L73

You can create a command that simply opens your view when executed, and add it to your custom menu.

**Reply by vince-fugnitto** [reply]
You can take a look how the `getting-started` widget is added to the `help` menu for an example as well:

https://github.com/eclipse-theia/theia/blob/31f78efffccc739e8497e43e2473aaf55e7fccd8/packages/getting-started/src/browser/getting-started-contribution.ts#L59-L72

**Reply by sboudreaux96** [reply]
with these methods I can just give the id of my widget within the custom menu (as shown in the first example you gave) and it would work? I am just making sure because when I create a widget it uses super.registerMenus(menus) within the registerMenus() function

**Reply by vince-fugnitto** [reply]
You have everything to try :) If there are issues you can debug through it but it should be straightforward as in the example.
